dist/index.d.ts

```diff
@@ -2174,12 +2174,15 @@ type FIND_RUINS = 123;
 
 // Filter Options
 
-interface FilterOptions<T extends FindConstant, S extends FindTypes[T] = FindTypes[T]> {
-    filter: FilterFunction<FindTypes[T], S> | FilterObject | string;
+interface PredicateFilterOptions<T, S extends T> {
+    filter: PredicateFilterFunction<T, S>;
+}
+interface FilterOptions<T> {
+    filter: FilterFunction<T> | FilterObject | string;
 }
 
-// We do not need to mark params as optional, because this is used for callback functions, whose params are always optional
-type FilterFunction<T, S extends T> = (object: T, index: number, collection: T[]) => object is S;
+type PredicateFilterFunction<T, S extends T> = (object: T, index: number, collection: T[]) => object is S;
```

Comment:
I think this is redundant. You can just pass `T` in for the case where `S` is not given.

Reply:
> I think this is redundant. You can just pass `T` in for the case where `S` is not given.

I think leaving `FilterFunction` is more about making sure that just any lodash filter (that returns `unknown` may be used).

Reply:
That should be unrelated :)

Reply:
I have merged `PredicateFilterOption` into `FilterOption` now